include/convex_bodies/hpolytope.h

```diff
@@ -638,6 +639,213 @@ class HPolytope {
+            {
+                lamda = pick_first_intersection_time_with_boundary(lamda1, lamda2, i, facet_prev);
+                if (lamda < min_plus && lamda > 0) 
+                {
+                    min_plus = lamda;
+                    facet = i;
+                }
+            }
+            Av_data++;
+            sum_nom_data++;
+            Ac_data++;
+        }
+        facet_prev = facet;
+        return std::make_pair(min_plus, facet);
+    }
+
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r, //current poistion
+                                                    Point const& v, // current velocity
+                                                    VT& Ac,  // the product Ac where c is the bias vector of the exponential distribution
+                                                    NT const& T, // the variance of the exponential distribution
+                                                    VT& Ar, // the product Ar 
+                                                    VT& Av, // the product Av
+                                                    NT const& lambda_prev, // the intersection time of the previous reflection
+                                                    int& facet_prev) const //the facet that the trajectory hit in the previous reflection
+    {
+        NT lamda = 0;
+        NT lamda2 =0;
+        NT lamda1 =0;
+        NT alpha;
+        NT min_plus  = std::numeric_limits<NT>::max();
+        VT sum_nom;
+        unsigned int j;
+        int m = num_of_hyperplanes();
+        int facet = -1;
+
+        Ar.noalias() += ((lambda_prev * lambda_prev) / (-2.0*T)) * Ac + lambda_prev * Av;
+        sum_nom = Ar - b;
+        Av.noalias() = A * v.getCoefficients();
+
+        NT* sum_nom_data = sum_nom.data();
+        NT* Av_data = Av.data();
+        NT* Ac_data = Ac.data();
+
+        for (int i = 0; i < m; i++) 
+        {
+            alpha = -((*Ac_data) / (2.0 * T));
+            if (solve_quadratic_polynomial(alpha, (*Av_data), (*sum_nom_data), lamda1, lamda2)) 
+            {
+                lamda = pick_first_intersection_time_with_boundary(lamda1, lamda2, i, facet_prev);
+                if (lamda < min_plus && lamda > 0) 
+                {
+                    min_plus = lamda;
+                    facet = i;
+                }
+            }
+            Av_data++;
+            sum_nom_data++;
+            Ac_data++;
+        }
+        facet_prev = facet;
+        return std::make_pair(min_plus, facet);
+    }
+
+    NT pick_first_intersection_time_with_boundary(NT const& lamda1, NT const& lamda2, int const& current_facet, int const& previous_facet) const
+    {
+        if (lamda1 == lamda2)
+        {
+            return lamda1;
+        }
+        NT lamda;
+        const double tol = 1e-10;
+        if (lamda1 * lamda2 < NT(0)) 
+        {
+            if (previous_facet == current_facet) 
```

Comment:
this can be simplified 
```
lamda = previous_facet == current_facet 
    ? std::max(lamda1, lamda2) < NT(tol) ? std::min(lamda1, lamda2) : std::max(lamda1, lamda2)
    : std::max(lamda1, lamda2);
```

Reply:
Done, thanks for the code.